### ISSUE TITLE
support command substitution in "url" manifest's strings

### ISSUE THREAD

**Issue by bam80** [flatpak/flatpak#2433]
It would be useful e.g. for getting URL to latest release on Github:
https://developer.github.com/v3/repos/releases/#get-the-latest-release
https://stackoverflow.com/a/26738019/1063363

**Comment by bam80** [comment]
For now, I can see many module's sources just use git url's pointing to 'master', which is not good.
In the other hand, pointing to some static tag/branch is not always appropriate..

**Comment by TingPing** [comment]
Manifests are static data meant to be reproducible. This feature would be entirely counter to that goal.

**Comment by bam80** [comment]
But pointing to branch and not tag makes it not so static anyway?

**Comment by TingPing** [comment]
> But pointing to branch and not tag makes it not so static anyway?

Yes but this isn't acceptable for repos like Flathub and is only really used for development repos like `gnome-nightly`.

The thing is you are proposing this to be used for stable versions which makes no sense. Because it no longer becomes reproducable when you want it.

**Comment by bam80** [comment]
>The thing is you are proposing this to be used for stable versions which makes no sense

I didn't say that is for stable versions, no. That feature could be used inline with pointing to branch feature, for development and experimental builds.

**Comment by bam80** [comment]
Also, the build will still be reproducible:
>When flatpak-builder does the export, it also stores the manifest that was used for the build in /app/manifest.json. The manifest is 'resolved', i.e. git branch names are replaced by the actual commit IDs that were used in the build. 

http://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html
